packages/expo-cellular/android/src/main/java/expo/modules/cellular/CellularModule.kt

```diff
@@ -29,48 +31,69 @@ class CellularModule(private val mContext: Context) : ExportedModule(mContext),
     }
   }
 
-  @SuppressLint("MissingPermission")
+  @ExpoMethod
+  fun getCurrentCellularInfoAsync(promise: Promise) {
+    val telephonyManager =
+      (mContext.getSystemService(Context.TELEPHONY_SERVICE) as? TelephonyManager).takeIf {
+        it?.simState == TelephonyManager.SIM_STATE_READY
+      }
+    val bundle = Bundle().apply {
+      putBoolean("allowsVoip", SipManager.isVoipSupported(mContext))
+      putString("isoCountryCode", telephonyManager?.simCountryIso)
+      putString("carrier", telephonyManager?.simOperatorName)
+      putString("mobileCountryCode", telephonyManager?.simOperator?.substring(0, 3))
+      putString("mobileNetworkCode", telephonyManager?.simOperator?.substring(3))
+      putInt("generation", getCurrentGeneration(telephonyManager))
+    }
+    promise.resolve(bundle)
+  }
```

Comment:
Grouping these fields together doesn't seem like it provides a big benefit and makes it so that an app that wants to read just the country code now needs to try to get the cellular generation, which touches sensitive permissions and is going to throw a caught exception.

With separate methods, this issue goes away:
```
Cellular.allowsVoipAsync()
Cellular.getIsoCountryCodeAsync()
Cellular.getCarrierNameAsync()
Cellular.getMobileCountryCodeAsync()
Cellular.getMobileNetworkCodeAsync()
Cellular.getCellularGenerationAsync()
```